### ISSUE TITLE
are the parameters for 300 epoch with accuracy 71.4% same to the gitgub 1000 epoch?

### ISSUE THREAD

**Issue by Dongshengjiang** [facebookresearch/barlowtwins#9]
are the parameters for 300 epoch with accuracy 71.4% same to the gitgub 1000 epoch?

**Comment by jzbontar** [comment]
Yes, we used the same hyperparameters for the 300 epoch and the 1000 epoch training runs. For 300 epoch training you would run the following:

```
python main.py /path/to/imagenet/ --epochs 300 --batch-size 2048 --learning-rate 0.2 --lambd 0.0051 --projector 8192-8192-8192 --scale-loss 0.024
```

We will release a pre-trained model that was trained for 300 epochs shortly. We will include the training logs as well, so that it will be easier to reproduce.

**Comment by Dongshengjiang** [comment]
Thanks for your quick reply. I have another question. how are the values of lamda and loss_scale for projector dimension for 128,512,2048?

**Comment by jzbontar** [comment]
- **lambda**: I'm afraid we don't have a lot of data points for setting lambda with smaller projector dimensions. Sorry. 
- **loss_scale**: To match our "research" code you can use `196 / projector_dim` as the scaling factor for the loss. But these are by no means optimal values. We simply didn't know that scaling the loss matters when we were developing Barlow Twins and we were surprised when the open-source code (with unscaled loss) didn't work quite as well as our research code. We are actively looking into removing the scale loss parameter.

**Comment by Dongshengjiang** [comment]
Thanks for your reply, I think using the mean of both diag and off-diag may reduce the complex of searching a proper lambda and loss_scale for different projectior_dim.

**Comment by WOWNICE** [comment]
What's the rationale behind using this "loss scaler"? Is it equivalent to rescaling the learning rate?

**Comment by StphTphsn** [comment]
Yes. The LARS optimizer is invariant to this scaling factor, but in our current implementation the biases and batch normalization parameters are currently excluded from LARS adaptation, and so the effective learning rate for these parameters is affected by this scaling factor. We are working on a version which will not require this parameter.

**Comment by WOWNICE** [comment]
Yeah indeed, never notice this issue with LARS. Thank you so much for sharing this viewpoint!
